File: push/src/main/java/com/xiaomi/channel/commonutils/android/MIUIUtils.java
package com.xiaomi.channel.commonutils.android;

import android.content.Context;
import android.text.TextUtils;
import com.xiaomi.channel.commonutils.logger.MyLog;
import com.xiaomi.channel.commonutils.reflect.JavaCalls;
import com.xiaomi.push.service.PushConstants;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/android/MIUIUtils.class */
public class MIUIUtils {
    private static final String ANDROID_SYSTEM_PROPERTIES = "android.os.SystemProperties";
    public static final int IS_MIUI = 1;
    private static final String KEY_MIUI_VERSION_CODE = "ro.miui.ui.version.code";
    private static final String KEY_MIUI_VERSION_NAME = "ro.miui.ui.version.name";
    public static final String MIUI_OS_VERSION_ALPHA = "alpha";
    public static final String MIUI_OS_VERSION_DEVELOPMENT = "development";
    public static final String MIUI_OS_VERSION_STABLE = "stable";
    public static final int NOT_MIUI = 2;
    public static final int UNKNOWN = 0;
    private static int isMIUI = 0;
    private static int isInXMS = -1;
    private static Map<String, Region> locale2RegionMap = null;

    private MIUIUtils() {
    }

    private static Region findServerRegionByLocale(String str) {
        initLocale2RegionMap();
        return locale2RegionMap.get(str.toUpperCase());
    }

    public static String getCountryCode() {
        String str = SystemProperties.get("ro.miui.region", "");
        String str2 = str;
        if (TextUtils.isEmpty(str)) {
            str2 = SystemProperties.get("persist.sys.oppo.region", "");
        }
        String str3 = str2;
        if (TextUtils.isEmpty(str2)) {
            str3 = SystemProperties.get("ro.oppo.regionmark", "");
        }
        String str4 = str3;
        if (TextUtils.isEmpty(str3)) {
            str4 = SystemProperties.get("ro.hw.country", "");
        }
        String str5 = str4;
        if (TextUtils.isEmpty(str4)) {
            str5 = SystemProperties.get("ro.csc.countryiso_code", "");
        }
        String str6 = str5;
        if (TextUtils.isEmpty(str5)) {
            str6 = SystemProperties.get("ro.product.country.region", "");
        }
        String str7 = str6;
        if (TextUtils.isEmpty(str6)) {
            str7 = SystemProperties.get("gsm.vivo.countrycode", "");
        }
        String str8 = str7;
        if (TextUtils.isEmpty(str7)) {
            str8 = SystemProperties.get("persist.sys.oem.region", "");
        }
        String str9 = str8;
        if (TextUtils.isEmpty(str8)) {
            str9 = SystemProperties.get("ro.product.locale.region", "");
        }
        String str10 = str9;
        if (TextUtils.isEmpty(str9)) {
            str10 = SystemProperties.get("persist.sys.country", "");
        }
        if (!TextUtils.isEmpty(str10)) {
            MyLog.w("get region from system, region = " + str10);
        }
        String country = str10;
        if (TextUtils.isEmpty(str10)) {
            country = Locale.getDefault().getCountry();
            MyLog.w("locale.default.country = " + country);
        }
        return country;
    }

    public static int getIsMIUI() {
        int i;
        synchronized (MIUIUtils.class) {
            try {
                if (isMIUI == 0) {
                    try {
                        isMIUI = !TextUtils.isEmpty(getProperty(KEY_MIUI_VERSION_CODE)) || !TextUtils.isEmpty(getProperty(KEY_MIUI_VERSION_NAME)) ? 1 : 2;
                    } catch (Throwable th) {
                        MyLog.e("get isMIUI failed", th);
                        isMIUI = 0;
                    }
                    MyLog.i("isMIUI's value is: " + isMIUI);
                }
                i = isMIUI;
            } catch (Throwable th2) {
                throw th2;
            }
        }
        return i;
    }

    public static String getMIUIType() {
        synchronized (MIUIUtils.class) {
            try {
                int mIUIType = SystemUtils.getMIUIType();
                return (!isMIUI() || mIUIType <= 0) ? "" : mIUIType < 2 ? MIUI_OS_VERSION_ALPHA : mIUIType < 3 ? MIUI_OS_VERSION_DEVELOPMENT : MIUI_OS_VERSION_STABLE;
            } catch (Throwable th) {
                throw th;
            }
        }
    }

    public static int getMiuiVersionCode(Context context) {
        String property = getProperty(KEY_MIUI_VERSION_CODE);
        return (TextUtils.isEmpty(property) || !TextUtils.isDigitsOnly(property)) ? 0 : Integer.parseInt(property);
    }

    public static String getProperty(String str) {
        String str2;
        try {
            try {
                str2 = (String) JavaCalls.callStaticMethod(ANDROID_SYSTEM_PROPERTIES, "get", str, "");
            } catch (Exception e) {
                MyLog.e("fail to get property. " + e);
                str2 = null;
            }
            return str2;
        } catch (Throwable th) {
            return null;
        }
    }

    public static Region getRegion(String str) {
        Region regionFindServerRegionByLocale = findServerRegionByLocale(str);
        return regionFindServerRegionByLocale == null ? Region.Global : regionFindServerRegionByLocale;
    }

    private static void initLocale2RegionMap() {
        if (locale2RegionMap != null) {
            return;
        }
        HashMap<String, Region> map = new HashMap<>();
        locale2RegionMap = map;
        map.put("CN", Region.China);
        locale2RegionMap.put("FI", Region.Europe);
        locale2RegionMap.put("SE", Region.Europe);
        locale2RegionMap.put("NO", Region.Europe);
        locale2RegionMap.put("FO", Region.Europe);
        locale2RegionMap.put("EE", Region.Europe);
        locale2RegionMap.put("LV", Region.Europe);
        locale2RegionMap.put("LT", Region.Europe);
        locale2RegionMap.put("BY", Region.Europe);
        locale2RegionMap.put("MD", Region.Europe);
        locale2RegionMap.put("UA", Region.Europe);
        locale2RegionMap.put("PL", Region.Europe);
        locale2RegionMap.put("CZ", Region.Europe);
        locale2RegionMap.put("SK", Region.Europe);
        locale2RegionMap.put("HU", Region.Europe);
        locale2RegionMap.put("DE", Region.Europe);
        locale2RegionMap.put("AT", Region.Europe);
        locale2RegionMap.put("CH", Region.Europe);
        locale2RegionMap.put("LI", Region.Europe);
        locale2RegionMap.put("GB", Region.Europe);
        locale2RegionMap.put("IE", Region.Europe);
        locale2RegionMap.put("NL", Region.Europe);
        locale2RegionMap.put("BE", Region.Europe);
        locale2RegionMap.put("LU", Region.Europe);
        locale2RegionMap.put("FR", Region.Europe);
        locale2RegionMap.put("RO", Region.Europe);
        locale2RegionMap.put("BG", Region.Europe);
        locale2RegionMap.put("RS", Region.Europe);
        locale2RegionMap.put("MK", Region.Europe);
        locale2RegionMap.put("AL", Region.Europe);
        locale2RegionMap.put("GR", Region.Europe);
        locale2RegionMap.put("SI", Region.Europe);
        locale2RegionMap.put("HR", Region.Europe);
        locale2RegionMap.put("IT", Region.Europe);
        locale2RegionMap.put("SM", Region.Europe);
        locale2RegionMap.put("MT", Region.Europe);
        locale2RegionMap.put("ES", Region.Europe);
        locale2RegionMap.put("PT", Region.Europe);
        locale2RegionMap.put("AD", Region.Europe);
        locale2RegionMap.put("CY", Region.Europe);
        locale2RegionMap.put("DK", Region.Europe);
        locale2RegionMap.put("RU", Region.Russia);
        locale2RegionMap.put("IN", Region.India);
    }

    public static boolean isGlobalRegion() {
        return !Region.China.name().equalsIgnoreCase(getRegion(getCountryCode()).name());
    }

    public static boolean isMIUI() {
        boolean z;
        synchronized (MIUIUtils.class) {
            try {
                z = true;
                if (getIsMIUI() != 1) {
                    z = false;
                }
            } catch (Throwable th) {
                throw th;
            }
        }
        return z;
    }

    public static boolean isNotMIUI() {
        boolean z;
        synchronized (MIUIUtils.class) {
            try {
                z = getIsMIUI() == 2;
            } finally {
            }
        }
        return z;
    }

    public static boolean isXMS() {
        boolean z = true;
        if (isInXMS < 0) {
            Object objCallStaticMethod = JavaCalls.callStaticMethod("miui.external.SdkHelper", "isMiuiSystem", new Object[0]);
            isInXMS = 0;
            if (objCallStaticMethod != null && (objCallStaticMethod instanceof Boolean) && !((Boolean) Boolean.class.cast(objCallStaticMethod)).booleanValue()) {
                isInXMS = 1;
            }
        }
        if (isInXMS <= 0) {
            z = false;
        }
        return z;
    }

    public static boolean isXMSF(Context context) {
        return context != null && isXMSF(context.getPackageName());
    }

    public static boolean isXMSF(String str) {
        return PushConstants.PUSH_SERVICE_PACKAGE_NAME.equals(str);
    }
}

File: legacy/src/main/java/com/xiaomi/channel/commonutils/android/MIUIUtils.kt
package com.xiaomi.channel.commonutils.android

import android.content.Context
import android.text.TextUtils
import com.xiaomi.channel.commonutils.logger.MyLog
import com.xiaomi.channel.commonutils.reflect.JavaCalls
import java.util.Locale

/*
 * Current override reference: com.xiaomi.xmsf 0.3.17-20260410000745 (versionCode 1003003000),
 * base.apk sha256 f3d72b6f5e1427ceecd3147a051d58e4dc95bb528397d486658e01cad9f7e590,
 * JADX path: com.xiaomi.xmsf/current/base/sources/com/xiaomi/channel/commonutils/android/MIUIUtils.java
 */
object MIUIUtils {
    private const val ANDROID_SYSTEM_PROPERTIES = "android.os.SystemProperties"
    private const val XMSF_PACKAGE_NAME = "com.xiaomi.xmsf"
    const val IS_MIUI = 1
    private const val KEY_MIUI_VERSION_CODE = "ro.miui.ui.version.code"
    private const val KEY_MIUI_VERSION_NAME = "ro.miui.ui.version.name"
    const val MIUI_OS_VERSION_ALPHA = "alpha"
    const val MIUI_OS_VERSION_DEVELOPMENT = "development"
    const val MIUI_OS_VERSION_STABLE = "stable"
    const val NOT_MIUI = 2
    const val UNKNOWN = 0

    @Volatile
    private var isMIUI = UNKNOWN

    @Volatile
    private var isInXMS = -1
    private var locale2RegionMap: Map<String, Region>? = null

    private fun findServerRegionByLocale(value: String?): Region? {
        initLocale2RegionMap()
        return if (value == null) {
            null
        } else {
            locale2RegionMap?.get(value.uppercase(Locale.getDefault()))
        }
    }

    @JvmStatic
    fun getCountryCode(): String {
        val regionKeys = arrayOf(
            "ro.miui.region",
            "persist.sys.oppo.region",
            "ro.oppo.regionmark",
            "ro.hw.country",
            "ro.csc.countryiso_code",
            "ro.product.country.region",
            "gsm.vivo.countrycode",
            "persist.sys.oem.region",
            "ro.product.locale.region",
            "persist.sys.country",
        )
        var country = ""
        for (key in regionKeys) {
            country = SystemProperties.get(key, "")
            if (!TextUtils.isEmpty(country)) {
                break
            }
        }
        if (!TextUtils.isEmpty(country)) {
            MyLog.w("get region from system, region = $country")
        }
        if (TextUtils.isEmpty(country)) {
            country = Locale.getDefault().country
            MyLog.w("locale.default.country = $country")
        }
        return country
    }

    @JvmStatic
    @Synchronized
    fun getIsMIUI(): Int {
        if (isMIUI == UNKNOWN) {
            try {
                isMIUI = if (
                    !TextUtils.isEmpty(getProperty(KEY_MIUI_VERSION_CODE)) ||
                    !TextUtils.isEmpty(getProperty(KEY_MIUI_VERSION_NAME))
                ) {
                    IS_MIUI
                } else {
                    NOT_MIUI
                }
            } catch (throwable: Throwable) {
                MyLog.e("get isMIUI failed", throwable)
                isMIUI = UNKNOWN
            }
            MyLog.i("isMIUI's value is: $isMIUI")
        }
        return isMIUI
    }

    @JvmStatic
    @Synchronized
    fun getMIUIType(): String {
        val miuiType = SystemUtils.getMIUIType()
        return if (!isMIUI() || miuiType <= 0) {
            ""
        } else if (miuiType < 2) {
            MIUI_OS_VERSION_ALPHA
        } else if (miuiType < 3) {
            MIUI_OS_VERSION_DEVELOPMENT
        } else {
            MIUI_OS_VERSION_STABLE
        }
    }

    @JvmStatic
    @Suppress("UNUSED_PARAMETER")
    fun getMiuiVersionCode(context: Context): Int {
        val property = getProperty(KEY_MIUI_VERSION_CODE)
        return if (TextUtils.isEmpty(property) || !TextUtils.isDigitsOnly(property)) {
            0
        } else {
            property!!.toInt()
        }
    }

    @JvmStatic
    fun getProperty(str: String): String? {
        return try {
            try {
                JavaCalls.callStaticMethod(ANDROID_SYSTEM_PROPERTIES, "get", str, "") as? String
            } catch (e: Exception) {
                MyLog.e("fail to get property. $e")
                null
            }
        } catch (_: Throwable) {
            null
        }
    }

    @JvmStatic
    fun getRegion(str: String?): Region {
        return findServerRegionByLocale(str) ?: Region.Global
    }

    private fun initLocale2RegionMap() {
        if (locale2RegionMap != null) {
            return
        }
        locale2RegionMap = hashMapOf(
            "CN" to Region.China,
            "FI" to Region.Europe,
            "SE" to Region.Europe,
            "NO" to Region.Europe,
            "FO" to Region.Europe,
            "EE" to Region.Europe,
            "LV" to Region.Europe,
            "LT" to Region.Europe,
            "BY" to Region.Europe,
            "MD" to Region.Europe,
            "UA" to Region.Europe,
            "PL" to Region.Europe,
            "CZ" to Region.Europe,
            "SK" to Region.Europe,
            "HU" to Region.Europe,
            "DE" to Region.Europe,
            "AT" to Region.Europe,
            "CH" to Region.Europe,
            "LI" to Region.Europe,
            "GB" to Region.Europe,
            "IE" to Region.Europe,
            "NL" to Region.Europe,
            "BE" to Region.Europe,
            "LU" to Region.Europe,
            "FR" to Region.Europe,
            "RO" to Region.Europe,
            "BG" to Region.Europe,
            "RS" to Region.Europe,
            "MK" to Region.Europe,
            "AL" to Region.Europe,
            "GR" to Region.Europe,
            "SI" to Region.Europe,
            "HR" to Region.Europe,
            "IT" to Region.Europe,
            "SM" to Region.Europe,
            "MT" to Region.Europe,
            "ES" to Region.Europe,
            "PT" to Region.Europe,
            "AD" to Region.Europe,
            "CY" to Region.Europe,
            "DK" to Region.Europe,
            "RU" to Region.Russia,
            "IN" to Region.India,
        )
    }

    @JvmStatic
    fun isGlobalRegion(): Boolean {
        return !Region.China.name.equals(getRegion(getCountryCode()).name, ignoreCase = true)
    }

    @JvmStatic
    @Synchronized
    fun isMIUI(): Boolean {
        return getIsMIUI() == IS_MIUI
    }

    @JvmStatic
    @Synchronized
    fun isNotMIUI(): Boolean {
        return getIsMIUI() == NOT_MIUI
    }

    @JvmStatic
    fun isXMS(): Boolean {
        if (isInXMS < 0) {
            val result = JavaCalls.callStaticMethod("miui.external.SdkHelper", "isMiuiSystem")
            isInXMS = 0
            if (result is Boolean && !result) {
                isInXMS = 1
            }
        }
        return isInXMS > 0
    }

    @JvmStatic
    fun isXMSF(context: Context?): Boolean {
        return context != null && isXMSF(context.packageName)
    }

    @JvmStatic
    fun isXMSF(str: String?): Boolean {
        return XMSF_PACKAGE_NAME == str
    }
}
